messaging/src/main/java/org/axonframework/messaging/MessageStream.java

```diff
@@ -111,106 +212,123 @@ static <M extends Message<?>> MessageStream<M> just(M instance) {
      * <p>
-     * Note that parallel processes <b>are not</b> supported!
+     * Note that parallel processing <b>is not</b> supported!
      *
      * @param identity    The initial value given to the {@code accumulator}.
-     * @param accumulator The {@link BiFunction} accumulating all {@link Message Messages} from this
+     * @param accumulator The {@link BiFunction} accumulating all {@link Entry entries} from this
      *                    {@link MessageStream stream} into a return value of type {@code R}.
-     * @param <R>         The result of the {@code accumulator} after reducing all {@link Message Messages} from this
+     * @param <R>         The result of the {@code accumulator} after reducing all {@link Entry entries} from this
      *                    {@link MessageStream stream}.
      * @return A {@link CompletableFuture} carrying the result of the given {@code accumulator} that reduced the entire
      * {@link MessageStream stream}.
      */
-    <R> CompletableFuture<R> reduce(@NotNull R identity,
-                                    @NotNull BiFunction<R, M, R> accumulator);
+    <R> CompletableFuture<R> reduce(@Nonnull R identity,
+                                    @Nonnull BiFunction<R, Entry<M>, R> accumulator);
 
     /**
-     * Invokes the given {@code onNext} each time a {@link Message} is consumed from this {@link MessageStream stream}.
+     * Invokes the given {@code onNext} each time an {@link Entry entry} is consumed from this
+     * {@link MessageStream stream}.
      * <p>
-     * Depending on the stream's implementation, the function may be invoked when the item is provided to the
+     * Depending on the stream's implementation, the function may be invoked when the entry is provided to the
      * {@link Consumer}, or at the moment it's available for reading on the stream. Subscribing multiple times to the
-     * resulting stream may cause the given {@code onNext} to be invoked more than once for an item.
+     * resulting stream may cause the given {@code onNext} to be invoked more than once for an entry.
      *
-     * @param onNext The {@link Consumer} to invoke for each item.
-     * @return A {@link MessageStream stream} that will invoke the given {@code onNext} for each item.
+     * @param onNext The {@link Consumer} to invoke for each {@link Entry entry}.
+     * @return A {@link MessageStream stream} that will invoke the given {@code onNext} for each {@link Entry entry}.
      */
-    default MessageStream<M> onNextItem(@NotNull Consumer<M> onNext) {
+    default MessageStream<M> onNextItem(@Nonnull Consumer<Entry<M>> onNext) {
         return new OnNextMessageStream<>(this, onNext);
     }
 
     /**
      * TODO this is by no means done yet...please fix.
-     *
-     * @param consumer
-     * @return
      */
-    default MessageStream<M> consume(@NotNull Predicate<M> consumer) {
+    default MessageStream<M> consume(@Nonnull Predicate<Entry<M>> consumer) {
```

Comment:
If you don't mind, I'll drop the `consume` operation entirely. This is a leftover brainfart with no use right now.
If we need it, it will be introduced at that point in time.